django/apps/registry.py

```diff
@@ -69,53 +70,58 @@ def populate(self, installed_apps=None):
         # populate() might be called by two threads in parallel on servers
         # that create threads before initializing the WSGI callable.
         with self._lock:
-            if self.ready:
-                return
-
-            # app_config should be pristine, otherwise the code below won't
-            # guarantee that the order matches the order in INSTALLED_APPS.
-            if self.app_configs:
-                raise RuntimeError("populate() isn't reentrant")
-
-            # Phase 1: initialize app configs and import app modules.
-            for entry in installed_apps:
-                if isinstance(entry, AppConfig):
-                    app_config = entry
-                else:
-                    app_config = AppConfig.create(entry)
-                if app_config.label in self.app_configs:
+            try:
```

Comment:
It's unclear to me why you need this `try / finally` block.

If an exception happens `populate()` in an "inner" call — and any inner call will raise an exception just below — `_populate_running` shouldn't be reset to `False` until the "outer" call terminates.

In fact I don't think we need to reset `_populate_running` at all. I think it will be just as easy to make it a simple boolean that starts at `False` and moves to `True` when `populate()` is called for the first time. That will be consistent with `(apps_|models_|)ready`. I'd just call it `loading` or `started` and not bother reset it.